de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/server/rpl/exp/RabbitMqExportProcessor.java

```diff
@@ -0,0 +1,195 @@
+import org.compiere.model.I_EXP_ProcessorParameter;
+import org.compiere.model.MEXPProcessor;
+import org.compiere.model.X_EXP_ProcessorParameter;
+import org.compiere.util.Trx;
+import org.slf4j.Logger;
+import org.springframework.amqp.core.BindingBuilder;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.core.TopicExchange;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.w3c.dom.Document;
+
+import javax.xml.transform.*;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.Properties;
+
+/**
+ * Send AMQP Messages
+ */
+public class RabbitMqExportProcessor implements IExportProcessor
+{
+
+	/**
+	 * Name of the exchange name parameter
+	 */
+	private static final String EXCHANGE_NAME_PARAMETER = "exchangeName";
+	/**
+	 * Name of the routing key parameter
+	 */
+	private static final String ROUTING_KEY_PARAMETER = "routingKey";
+	/**
+	 * Name of the is durable queue parameter
+	 */
+	private static final String IS_DURABLE_QUEUE_PARAMETER = "isDurableQueue";
+	/**
+	 * Logger
+	 */
+	protected Logger log = LogManager.getLogger(getClass());
+
+	@Override
+	public void process(Properties ctx, MEXPProcessor expProcessor, Document document, Trx trx)
+			throws Exception
+	{
+		String host = expProcessor.getHost();
+		int port = expProcessor.getPort();
+		String account = expProcessor.getAccount();
+		String password = expProcessor.getPasswordInfo();
+		String exchangeName = StringUtils.EMPTY;
+		String routingKey = StringUtils.EMPTY;
+		boolean isDurableQueue = true;
+
+		// Read all processor parameters and set them!
+		I_EXP_ProcessorParameter[] processorParameters = expProcessor.getEXP_ProcessorParameters();
+		if (processorParameters != null && processorParameters.length > 0)
+		{
+			for (I_EXP_ProcessorParameter processorParameter : processorParameters)
+			{
+				log.info("ProcesParameter          Value = " + processorParameter.getValue());
+				log.info("ProcesParameter ParameterValue = " + processorParameter.getParameterValue());
+				if (processorParameter.getValue().equals(EXCHANGE_NAME_PARAMETER))
+				{
+					exchangeName = processorParameter.getParameterValue();
+				}
+				else if (processorParameter.getValue().equals(ROUTING_KEY_PARAMETER))
+				{
+					routingKey = processorParameter.getParameterValue();
+				}
+				else if (processorParameter.getValue().equals(IS_DURABLE_QUEUE_PARAMETER))
+				{
+					isDurableQueue = Boolean.parseBoolean(processorParameter.getParameterValue());
+				}
+			}
+		}
+
+		if (exchangeName == null || exchangeName.length() == 0)
+		{
+			throw new Exception("Missing " + X_EXP_ProcessorParameter.Table_Name + " with key 'exchangeName'!");
+		}
+
+		if (routingKey == null || routingKey.length() == 0)
```

Comment:
use `Check.isEmpty(routingKey, true);` instead.
Or consider replacing the whole block with `Check.assumeNotEmpty(routingKey, ThatNewExportProcessorException.class, "Missing {} with key 'routingKey'!", X_EXP_ProcessorParameter.Table_Name)`

(`de.metas.util.Check`)